### PR TITLE
Add TDX debugging tools and documentation

### PR DESCRIPTION
**Pull request by msanft**
This adds a suite of debugging utilities for low-level TDX problems such as mismatching RTMRs, as well as documentation on how to perform the debugging.

This includes:
- `tdx-tools` in our debug images, which is a Python utility to view RTMRs and their event log.
- A debug switch in `OVMF-TDX`, allowing to debug the TDX virtual firmware.
- A script to launch a TDX QEMU like Kata would, which can then be used with the debug build of OVMF, as Kata doesn't support it.

Additionally, some documentation is added on how to perform debugging on TDX measurements.

### CODE REVIEW

File: packages/by-name/OVMF-TDX/0001-verify-Hobs-instead-of-measuring-them.patch [hunk 1]
```diff
@@ -1,6 +1,6 @@
-From 649d2928a8f6552ba0644e6f9a7f719dfbdabe60 Mon Sep 17 00:00:00 2001
-From: Tom Dohrmann <erbse.13@gmx.de>
-Date: Mon, 23 Sep 2024 10:57:09 +0200
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Moritz Sanft <58110325+msanft@users.noreply.github.com>
```
Comment (burgerdev) [inline packages/by-name/OVMF-TDX/0001-verify-Hobs-instead-of-measuring-them.patch:2]: That's not supposed to happen, I believe.

**Review by burgerdev — APPROVED** [review]
Wow, thanks a lot!

File: packages/by-name/OVMF-TDX/0001-verify-Hobs-instead-of-measuring-them.patch (hunk 1 above)
Comment (msanft) [reply packages/by-name/OVMF-TDX/0001-verify-Hobs-instead-of-measuring-them.patch:2]: This is the result of recreation the patch in the form we advertise in `dev-docs/patches.md`

Comment (burgerdev) [reply packages/by-name/OVMF-TDX/0001-verify-Hobs-instead-of-measuring-them.patch:2]: `git am` should not replace the author :thinking:

Comment (msanft) [reply packages/by-name/OVMF-TDX/0001-verify-Hobs-instead-of-measuring-them.patch:2]: I had to `git apply --3way`, as `git am --3way` didn't work for some reason I don't remember anymore. Maybe this changed it, but I'll revert the author change manually for now.

**Pull request merged**